extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -9,161 +9,144 @@ class EmbeddableDiscussionsController {
 	const COLUMNS_MIN = 1;
 	const COLUMNS_MAX = 2;
 
+	const PARAM_MOSTRECENT = 'mostrecent';
+	const PARAM_SIZE = 'size';
+	const PARAM_COLUMNS = 'columns';
+	const PARAM_CATEGORY = 'category';
+
 	private $templateEngine;
 
-	function __construct() {
+	public function __construct() {
 		$this->templateEngine = ( new Wikia\Template\MustacheEngine )->setPrefix( __DIR__ . '/templates' );
 	}
 
-	public static function onParserFirstCallInit( Parser $parser ) {
-		global $wgEnableDiscussions;
+	public function render( array $args ) {
+		global $wgCityId;
 
-		if ( $wgEnableDiscussions ) {
-			$parser->setHook( self::TAG_NAME, [ new self(), 'render' ] );
-		}
+		$parameters = [
+			static::PARAM_MOSTRECENT => $args[ static::PARAM_MOSTRECENT ] ?? false,
+			static::PARAM_SIZE => $args[ static::PARAM_SIZE ] ?? self::ITEMS_DEFAULT,
+			static::PARAM_COLUMNS => $args[ static::PARAM_COLUMNS ] ?? self::COLUMNS_DEFAULT,
+			static::PARAM_CATEGORY => $args[ static::PARAM_CATEGORY ] ?? '',
+		];
 
-		return true;
-	}
+		$errorMessage = $this->validateArguments( $parameters );
```

Comment:
I still think this code needs to return ALL error messages from the UX perspective